serializers.py

```diff
@@ -58,7 +64,28 @@ def get_is_completed(self, instance):
 
             return instance.has_already_ended
 
+        def get_status_type(self, instance):
+            """
+            Returns status type for instance 
+            :return: if the duration of an instance has ended
+            """
+
+            return "A" if instance.has_already_ended else "O"
+
+        def get_semester_name(self, instance):
+            """
+            Returns semester name from semester annotations. i.e Spring for s.
+            :return: semester name from semester annotations
+            """
+
+            for sem in SEMESTER_TYPES:
```

Comment:
Django has a function called `get_semester_display()` (get_choice field name_display) which can fetch you the display text of the choices.

Reply:
I would also prefer if this function can be written in `cms-omniport-api` instead.